### PR TITLE
Electron update

### PR DESCRIPTION
**Pull request by rmcvey**
This PR includes many changes to Stethoscope:
- Updated Electron to v5 (from recently deprecated v2)
- Fixed electron migration issues
- Migrated Node/Electron code to ES6
- Changed the way `kmd` scripts are executed, now on-demand, rather than all checks on each query
- Added encryption check and flag to all connected disks
- Changed `resolvers/platform` to export the relevant resolvers for the current platform
- Added `/raw` endpoint that returns the raw `kmd` response from all checks
- Added `kmd` command `boolMatch` that sets `true`|`false` based on matching

Also started work on adding support for application querying.

This requires a `kmd` [PR merge](https://github.com/Netflix-Skunkworks/kmd/pull/8) and publish

### CODE REVIEW

File: src/Menu.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ module.exports = function (mainWindow, app, focusOrCreateWindow, updater, log) {
     }, {
       label: 'Copy Debug Info',
       click () {
-        fetch(`http://127.0.0.1:37370/debugger`, {
+        window.fetch(`http://127.0.0.1:37370/debugger`, {
```
Comment (nathancharles) [inline src/Menu.js]: It might be better to make fetch an allowed global, in any case that window is undefined

File: src/resolvers/Device.js [hunk 2]
```diff
@@ -25,73 +26,82 @@ const securityToPassFailStatus = status => {
...
-  deviceName (root, args, { kmdResponse }) {
-    return kmdResponse.system.hostname
+  async deviceName (root, args, context) {
+    const result = await kmd('hostname', context)
+    return result.system.hostname
   },
 
   platform (root, args, context) {
-    return context.platform
+    return (context && context.platform) || process.platform
```
Comment (nathancharles) [inline src/resolvers/Device.js]: nit: you could use a default param for context and simplify the check here

File: src/start.js [hunk 3]
```diff
@@ -14,33 +14,37 @@
...
-const findIcon = require('./lib/findIcon')(env)
-const startGraphQLServer = require('../server')
+const findIcon = iconFinder(env)
 const IS_DEV = env === 'development'
-const { IS_MAC, IS_WIN } = require('./lib/platform')
-const AutoLauncher = require('./AutoLauncher')
-
 const disableAutomaticScanning = settings.get('disableAutomaticScanning')
 
+scripts.init()
```
Comment (nathancharles) [inline src/start.js]: This shouldn't be needed since we are running the scripts as needed, correct?

Comment (rmcvey) [reply src/start.js]: This just preloads the scripts so that they can be executed, it does not execute any scripts.

Comment (nathancharles) [reply src/start.js]: Does precompiling still make sense as we are compiling and running on each kmd call?

**Pull request merged**